packages/hash/frontend/src/lib/org.ts

```diff
@@ -51,26 +53,76 @@ export const constructMinimalOrg = (params: {
 };
 
 export type Org = MinimalOrg & {
-  members: (MinimalUser & { responsibility: string })[];
+  members: (User & { responsibility: string })[];
 };
 
 export const constructOrg = (params: {
   subgraph: Subgraph;
   orgEntityEditionId: EntityEditionId;
+  resolvedUsers?: Record<EntityId, User>;
+  resolvedOrgs?: Record<EntityId, Org>;
 }): Org => {
   const { subgraph, orgEntityEditionId } = params;
 
-  return {
-    ...constructMinimalOrg({
+  const resolvedUsers = params.resolvedUsers ?? {};
+  const resolvedOrgs = params.resolvedOrgs ?? {};
+
+  const orgMemberships = getIncomingLinksForEntityAtMoment(
+    subgraph,
+    orgEntityEditionId.baseId,
+    new Date(),
+  ).filter(
+    (linkEntity) =>
+      linkEntity.metadata.entityTypeId ===
+      types.linkEntityType.orgMembership.linkEntityTypeId,
+  );
+
+  const org = constructMinimalOrg({
+    subgraph,
+    orgEntityEditionId,
+  }) as Org;
+
+  // We add it to resolved orgs *before* fully creating so that when we're traversing we know
+  // we already encountered it and avoid infinite recursion
+  resolvedOrgs[org.entityEditionId.baseId] = org;
+
+  org.members = orgMemberships.map(({ metadata, properties }) => {
+    const responsibility: string = properties[
+      extractBaseUri(types.propertyType.responsibility.propertyTypeId)
+    ] as string;
+
+    if (!metadata.linkMetadata?.leftEntityId) {
+      throw new Error("Expected org membership to contain a left entity");
+    }
+    const userEntity = getLeftEntityForLinkEntityAtMoment(
       subgraph,
-      orgEntityEditionId,
-    }),
-    /**
-     * @todo implement members once we are able to fetch incoming links
-     *   see https://app.asana.com/0/1201095311341924/1203250435416416/f
-     */
-    members: [],
-  };
-};
+      metadata.editionId.baseId,
+      new Date(),
+    );
+
+    let user = resolvedUsers[userEntity.metadata.editionId.baseId];
+
+    if (
+      !(
+        user &&
+        user.entityEditionId.version >= userEntity.metadata.editionId.version
+      )
```

Comment:
This deduplication is temporary I assume? If not, it would be nice that we do the same for the orgs themselves and not just the users here (i am aware we also deduplicate the orgs in the `MemberOf` field of the user :) )

Reply:
It's not really deduplication so much as making sure the resolved user is the most up-to-date one. There is a problem I have realised now, if we're returning multiple versions of a user in the subgraph (or of an org), then if that is encountered earlier in the traversal, some of the references in the created objects won't be updated when replacing the one in the map.. hm.

This is one of those places where having more explicit references vs values would be helpful

Reply:
The method was operating over `EditionId`s anyway, so I shouldn't have been discarding non-latest entities. I've reworked this, thanks for bringing up a comment.